src/ray/object_manager/object_store_notification_manager.cc

```diff
@@ -63,25 +63,25 @@ void ObjectStoreNotificationManager::ProcessStoreNotification(
 }
 
 void ObjectStoreNotificationManager::ProcessStoreAdd(const ObjectInfoT &object_info) {
-  for (auto handler : add_handlers_) {
+  for (auto& handler : add_handlers_) {
     handler(object_info);
   }
 }
 
 void ObjectStoreNotificationManager::ProcessStoreRemove(const ObjectID &object_id) {
-  for (auto handler : rem_handlers_) {
+  for (auto& handler : rem_handlers_) {
```

Comment:
`add_handlers_` and `rem_handlers_` are vectors of function pointers. Did you find switching to a reference giving you a significant performance improvement?

Reply:
std::function copy is not so cheap, the std::function may be a wrapper of lambda. In the lambda, it may catch some variables, it need copy of the object. For it you can debug how lambda was converted to a std::function.

Reply:
sounds good, I agree with that and have been pushing internally to use refs and const refs as much as possible... Just curious if you actually saw performance differences in your testing environment.

Reply:
I can't get the perf number now. Because the current code has too many difference from our internal branch.
I did such change because I met similar problem before.